src/types/transformation.ts

```diff
@@ -0,0 +1,212 @@
+  metadata: Metadata;
+  statusCode: number;
+  error?: string;
+  statTags?: object;
+};
+
+/**
+ * Router transformation structures
+ */
+export type RouterTransformationRequestData<
+  Message = object,
+  DestinationType = Destination,
+  ConnectionType = Connection,
+> = {
+  request?: object;
+  message: Message;
+  metadata: Metadata;
+  destination: DestinationType;
+  connection?: ConnectionType;
+};
+
+export type RouterTransformationRequest = {
+  input: RouterTransformationRequestData[];
+  destType: string;
+};
+
+export type RouterTransformationResponse = {
+  batchedRequest?: ProcessorTransformationOutput | ProcessorTransformationOutput[];
+  metadata: Metadata[];
+  destination: Destination;
+  batched: boolean;
+  statusCode: number;
+  error?: string;
+  statTags?: object;
+};
+
+/**
+ * Metadata structure for proxy requests
+ */
+export type ProxyMetdata = {
+  jobId: number;
+  attemptNum: number;
+  userId: string;
+  sourceId: string;
+  destinationId: string;
+  workspaceId: string;
+  secret: Record<string, unknown>;
+  destInfo?: Record<string, unknown>;
+  omitempty?: Record<string, unknown>;
+  dontBatch: boolean;
+};
+
+/**
+ * Proxy request structure for version 0
+ */
+export type ProxyV0Request = {
+  version: string;
+  type: string;
+  method: string;
+  endpoint: string;
+  userId: string;
+  headers?: Record<string, unknown>;
+  params?: Record<string, unknown>;
+  body?: {
+    JSON?: Record<string, unknown>;
+    JSON_ARRAY?: Record<string, unknown>;
+    XML?: Record<string, unknown>;
+    FORM?: Record<string, unknown>;
+  };
+  files?: Record<string, unknown>;
+  metadata: ProxyMetdata;
+  destinationConfig: Record<string, unknown>;
+};
+
+/**
+ * Proxy request structure for version 1
+ */
+export type ProxyV1Request = {
+  version: string;
+  type: string;
+  method: string;
+  endpoint: string;
+  userId: string;
+  headers?: Record<string, unknown>;
+  params?: Record<string, unknown>;
+  body?: {
+    JSON?: Record<string, unknown>;
+    JSON_ARRAY?: Record<string, unknown>;
+    XML?: Record<string, unknown>;
+    FORM?: Record<string, unknown>;
+  };
+  files?: Record<string, unknown>;
+  metadata: ProxyMetdata[];
+  destinationConfig: Record<string, unknown>;
+};
+
+export type ProxyRequest = ProxyV0Request | ProxyV1Request;
+
+/**
+ * Error detailer structure for proxy
+ */
+export type ErrorDetailer = {
+  module: string;
+  implementation: string;
+  feature: string;
+} & ErrorDetailerOptions;
+
+export type ErrorDetailerOptions = {
+  errorCategory?: string;
+  errorType?: string;
+  meta?: string;
+  destType?: string;
+  srcType?: string;
+  cluster?: string;
+  customer?: string;
+  destinationId?: string;
+  workspaceId?: string;
+  sourceId?: string;
+};
+
+export type MetaTransferObjectForProxy = {
+  metadata?: ProxyMetdata;
+  metadatas?: ProxyMetdata[];
+  errorDetails: ErrorDetailer;
+  errorContext: string;
+};
+
+export type MetaTransferObject =
+  | {
```

Comment:
```suggestion
  {
```
does this work ?

Reply:
Yeah it works its just a formatting thing. On saving prettier automatically adding the `|` in front based on typescript configs